docs/src/use/migrate-to-9.0.0.md

```diff
@@ -553,6 +553,7 @@ In order to aid in the development of high-quality custom rules that are free fr
 1. **Suggestions must generate valid syntax.** In order for rule suggestions to be helpful, they need to be valid syntax. `RuleTester` now parses the output of suggestions using the same language options as the `code` value and throws an error if parsing fails.
 1. **Test cases must be unique.** Identical test cases can cause confusion and be hard to detect manually in a long test file. Duplicates are now automatically detected and can be safely removed.
 1. **`filename` and `only` must be of the expected type.** `RuleTester` now checks the type of `filename` and `only` properties of test objects. If specified, `filename` must be a string value. If specified, `only` must be a boolean value.
+1. **Messages cannot have unsubstituted placeholders.** The rule tester now checks if there are `{{ placeholder }}` still in the message as their values were not passed via `data` in the respective `context.report`.
```

Comment:
```suggestion
1. **Messages cannot have unsubstituted placeholders.** The rule tester now checks if there are {% raw %}`{{ placeholder }}`{% endraw %} still in the message as their values were not passed via `data` in the respective `context.report()` call.
```

We need to wrap this into `{% raw %}...{% endraw %}`, otherwise this syntax would have a meaning for tools we use to build the site. Here's how it currently looks in this PR's preview (note that the`{{ placeholder }}` text doesn't appear):

![image](https://github.com/eslint/eslint/assets/44349756/71c5de37-00d3-46f8-a936-6797cf097052)

Also, context.report -> context.report() call.

Also, can you add `[#18016](https://github.com/eslint/eslint/issues/18016)` to the list of related issues on line 560 (GitHub doesn't allow me to leave suggestions there)?